### ISSUE TITLE
chi.Walk missing routes

### ISSUE THREAD

**Issue by sauerbraten** [go-chi/chi#830]
Similarly to https://github.com/go-chi/chi/issues/750, I'm trying to use chi.Walk() to get a report of all routes. However, I found that some routes are missing when I use Route() as well as e.g. Get() with the same pattern: https://go.dev/play/p/7Ntz1yMoXrz

Interestingly, the route _is_ visited by the walk function when you change `r.Route("/bar", ...` to `r.Route("/bar/", ...`.

Since the `GET /foo/bar` request is handled in both cases, I also expect chi.Walk() to report the route in both cases.

**Comment by ganicus** [comment]
@sauerbraten did you find out anything else about this? I'm experiencing similar issues.

**Comment by sauerbraten** [comment]
No, I intend to replace my chi uses with https://pkg.go.dev/net/http@master#ServeMux once Go 1.22 is released.

**Comment by lai0xn** [comment]
i want to work on that

**Comment by VojtechVitek** [comment]
Fixed in https://github.com/go-chi/chi/pull/1148. Please review.
